homeassistant/components/shelly/__init__.py

```diff
@@ -169,7 +169,27 @@ async def _async_block_device_setup() -> None:
 
         platforms = BLOCK_SLEEPING_PLATFORMS
 
-        if not entry.data.get(CONF_SLEEP_PERIOD):
+        sleep_period = entry.data.get(CONF_SLEEP_PERIOD)
+
+        # Some old firmware have a wrong sleep period hardcoded value.
+        # Following code block will force the right value for affected devices
+        if sleep_period == 21600 and entry.data["model"] in [
+            MODEL_DW,
+            MODEL_DW_2,
+            MODEL_HT,
+        ]:
+            expected_sleep_period = 42300
+            LOGGER.warning(
```

Comment:
```suggestion
            LOGGER.info(
```

I think this can be in info level, no need to warn the user as there is no action needed from the user side we can silently update the value and it will still be visible if we ask for logs

Reply:
Many users have a default level of WARNING, this is why I prefer to make it visibile.
Anyway is a single shot, not a recurring one so if not visible the first time is lost.

Reply:
I think a warning should only be visible if there is something that the user needs to take care about (connection error, something that is configured incorrectly or doesn't work).

I had multiple times when I was asked in PRs to reduce the level not to make users worry about things they can't control or need to inform us.

In any case if another code-owner thinks this should be warning let's keep it as warning.